### ISSUE TITLE
when using plugins, only first build-block is run

### ISSUE THREAD

**Issue by rweng** [zont/gulp-usemin#96]
Hi, I got the following configuration:

index.html:
~~~
 <!-- build:css(.) styles/vendor.css -->
    <!-- bower:css -->
    <!-- endbower -->
    <!-- endbuild -->
    <!-- build:css(lib) styles/ext.css -->
    <link rel="stylesheet" href="ext-5.1.0/ext-theme-crisp-touch-all_01.css"/>
    <link rel="stylesheet" href="ext-5.1.0/ext-theme-crisp-touch-all_02.css"/>
    <!-- endbuild -->
    <!-- build:css styles/stylesheet.css -->
    <link rel="stylesheet" href="styles/stylesheet.css">
    <!-- endbuild -->
~~~

gulpfile.js
~~~
          .pipe($.usemin({
                // css: [$.minifyCss()],
               //  js: [$.uglify({mangle: false})],
                path: 'dist'
            }));
~~~

When both, `css` and `js` option are disabled, uglify generates all three stylesheets. However, when I enable the minification for css, only the first block is executed. I tried shifting the order and removing the bower block, but its always the first build block that is executed. Any idea why this is? shouldn't minifyCss() run over the same files as usemin? Btw, I also tried `['combine', $.minifyCss()]`.

Thanks, Robin

**Comment by kristijanhusak** [comment]
I'm having the same issue with uglify, all is good when i use only `usemin()`, but when i add `js:[uglify()]` it generates only first one.

**Comment by rweng** [comment]
@kristijanhusak just fyi, I switched to using `gulp-if` like that: `.pipe($.if('*.js', jsStream()))`

**Comment by pranaydutta89** [comment]
i am having the same issue ,is anybody aware of any alternative ,is this a usemin bug ??

**Comment by rweng** [comment]
@pranaydutta89 alternative is useref with gulp-uglify and gulp-if

**Comment by subaev** [comment]
same issue guys. works fine with plain usemin() w/o options but when i add any opts for css/html/js the first block only is generated.

**Comment by akullpp** [comment]
Same here.

**Comment by kristijanhusak** [comment]
@rweng
what exactly did you do in `jsStream()` function?

**Comment by rweng** [comment]
@kristijanhusak, basically something like this:

~~~js
function jsStream(){
         return lazypipe().pipe($.uglify, {mangle: false});
}
~~~

**Comment by kristijanhusak** [comment]
@rweng , thanks. I thought it is possible without additional plugins, but it requires lazypipe.
I handled it this similar to [this](https://github.com/yeoman/generator-gulp-webapp/blob/master/docs/recipes/asset-revisioning.md), because i need both `uglify()` and `rev()`

**Comment by zont** [comment]
Fixed in 0.3.14
